src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionValidator.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using static Hl7.Fhir.Model.Bundle;
+
+namespace Microsoft.Health.Fhir.Api.Features.Resources.Bundle
+{
+    public static class TransactionValidator
+    {
+        public static void ValidateTransaction(HashSet<string> resourceIdList, EntryComponent entry)
+        {
+            if (entry.Request.Method != HTTPVerb.GET)
+            {
+                string resourceId = GetResourceUrl(entry);
+
+                if (resourceIdList.Contains(resourceId))
+                {
+                    throw new RequestNotValidException(string.Format(Api.Resources.ResourcesMustBeUnique, resourceId));
+                }
+
+                resourceIdList.Add(resourceId);
+            }
+        }
+
+        private static string GetResourceUrl(EntryComponent component)
+        {
+            string fullUrl = component.FullUrl;
+
+            if (fullUrl != null)
+            {
+                if (fullUrl.StartsWith("urn", StringComparison.OrdinalIgnoreCase) || component.Request.Method == HTTPVerb.POST)
+                {
+                    return fullUrl;
+                }
+            }
+
+            return component.Request.Url;
```

Comment:
My understanding is fullURL will be present in POST. UUID will be used as a fullURL since the resource is not assigned a created yet and will not have the absoulute url/identity to represent it. According to spec "For transactions and batches, entries MAY not have fullURLs when the entry.request.method = POST, and the resource has no identity. Note that even in this case, there may still be a fullURL in a transaction on a POST so that relationships between resources can be represented"

Reply:
It will now return fullURl only for the valid post requests. Valid here means if the post is not for operations and also it will ignore if resourcetype is bundle.